### PR TITLE
Virtual Device Driver Sanity check

### PR DESCRIPTION
**Pull request by ClassicalComputer**
Hi Jonathan,

Here is my first pull request. 

This is not for merging into the Aporta developer repository, but just for feedback.

Lots of praying on my part to be able to understand the Aporta code base and make changes! So far, things are coming along well enough, and I have been able to look at your existing code to understand where to make the changes.

The virtual device driver configuration razor page currently has a badge number ("12345") hard coded for testing.

I spent the last couple of hours today struggling to try to figure out how to reference a textbox from Blazor, to be able to determine the badge number keyed in by the user on the Virtual driver configuration page! I still don't have a working solution to that.

After lots of searching on the internet, one solution I tried was referencing Microsoft.JSInterop. This library is supposed to let you reference the webpage DOM. My first attempts using JSInterop did not work. Do you know of a better solution?

Thank you so much again! I am really enjoying working on this project!

- Chris

### CODE REVIEW

File: src/Drivers/Aporta.Drivers.Virtual/VirtualDriver.cs [hunk 1]
```diff
@@ -71,7 +75,53 @@ public string ScrubSensitiveConfigurationData(string jsonConfigurationString)
...
+        var badgeAction = JsonConvert.DeserializeObject<BadgeSwipeAction>(parameters);
+        if (badgeAction == null) return;
+
+        _logger.LogInformation("A card has been presented to the reader");
+        var accessPoint = _endpoints.Where(endpoint => endpoint is IAccess).Cast<IAccess>()
+            .SingleOrDefault(accessPoint =>
+                $"VR{badgeAction.ReaderNumber}" == accessPoint.Id);
+
+
+        var cardID = badgeAction.CardData.ToBitArray();
```
Comment (bytedreamer) [inline src/Drivers/Aporta.Drivers.Virtual/VirtualDriver.cs:118]: No reason to create a local variable that is will be passed directly to the invoke method

File: src/Drivers/Aporta.Drivers.Virtual/VirtualDriver.cs [hunk 2]
```diff
@@ -102,4 +152,24 @@ protected virtual void OnOnlineStatusChanged(OnlineStatusChangedEventArgs eventA
     {
         OnlineStatusChanged?.Invoke(this, eventArgs);
     }
+
+    internal class TestCredentialReceivedHandler(BitArray cardData) : ICredentialReceivedHandler
```
Comment (bytedreamer) [inline src/Drivers/Aporta.Drivers.Virtual/VirtualDriver.cs:156]: Suggest naming it VirtualCredentialReceivedHandler. I thought it was a unit test helper class

**Comment by bytedreamer** [comment]
Looks good, added a couple of comments

**Comment by bytedreamer** [comment]
You want to bind a member variable in the text box. Checkout the binding section on this page

https://blazorise.com/docs/components/text

Also you can validate the text using the validation component

https://blazorise.com/docs/components/validation

File: src/Drivers/Aporta.Drivers.Virtual/VirtualDriver.cs (hunk 1 above)
Comment (ClassicalComputer) [reply src/Drivers/Aporta.Drivers.Virtual/VirtualDriver.cs:118]: Thank you. That is cleaner code. I made the change and will be checking in shortly.

File: src/Drivers/Aporta.Drivers.Virtual/VirtualDriver.cs (hunk 2 above)
Comment (ClassicalComputer) [reply src/Drivers/Aporta.Drivers.Virtual/VirtualDriver.cs:156]: Thank you. In all the anticipation of hoping to get the first version of the Virtual Device driver working I copied and pasted from the TestDriver and forgot to rename. Sorry about that. Thank you! Change is made and will be committed shortly.

**Comment by ClassicalComputer** [comment]
Thank you, Jonathan. I am now data binding the TextEdit boxes on the Virtual Driver configuration razor page. I am storing the bindings in a Dictionary, so that the code will work when there is more than one reader set up for the Virtual Device Driver.

**Pull request merged**